geomstats/special_orthogonal_group.py

```diff
@@ -480,165 +526,243 @@ def matrix_from_quaternion(self, quaternion):
         assert rot_mat.ndim == 3
         return rot_mat
 
-    def compose(self, point_1, point_2):
+    def compose(self, point_1, point_2, point_representation=None):
         """
         Compose two elements of SO(n).
         """
         point_1 = self.regularize(point_1)
         point_2 = self.regularize(point_2)
 
-        rot_mat_1 = self.matrix_from_rotation_vector(point_1)
-        rot_mat_2 = self.matrix_from_rotation_vector(point_2)
+        if point_representation is None:
+            point_representation = self.point_representation
 
-        rot_mat_prod = gs.einsum('ijk,ikl->ijl', rot_mat_1, rot_mat_2)
-        point_prod = self.rotation_vector_from_matrix(rot_mat_prod)
+        if point_representation == 'vector':
+            point_1 = self.matrix_from_rotation_vector(point_1)
+            point_2 = self.matrix_from_rotation_vector(point_2)
 
-        point_prod = self.regularize(point_prod)
+        point_prod = gs.einsum('ijk,ikl->ijl', point_1, point_2)
+
+        if point_representation == 'vector':
+            point_prod = self.rotation_vector_from_matrix(point_prod)
+
+        point_prod = self.regularize(
+            point_prod, point_representation=point_representation)
         return point_prod
 
-    def inverse(self, point):
+    def inverse(self, point, point_representation=None):
         """
         Compute the group inverse in SO(n).
         """
-        if self.n == 3:
-            inv_point = -self.regularize(point)
-        else:
-            rot_mat = self.matrix_from_rotation_vector(point)
-            inv_rot_mat = gs.linalg.inv(rot_mat)
-            inv_point = self.rotation_vector_from_matrix(inv_rot_mat)
+
+        if point_representation is None:
+            point_representation = self.point_representation
+
+        if point_representation == 'vector':
+            if self.n == 3:
+                inv_point = -self.regularize(point)
+                return inv_point
+            else:
+                point = self.matrix_from_rotation_vector(point)
```

Comment:
shouldn't the case n == 3 be covered by this?

Reply:
Yes, but I distinguish the case n==3 for efficiency (instead of converting to matrix and back). Shouldn't I?